### DISCUSSION TITLE
Worktrees beyond read support

### DISCUSSION THREAD

**Discussion by p-bakker** [eclipse-jgit/jgit#70] [Q&A]
Hi,

Recently [worktrees read support](https://github.com/eclipse-jgit/jgit/commit/f9beeb3b33e438fa3a10eb3871206573ae986abb) got added, which is a great step forwards!

My question is whether there's a list of known issues that need to be tackled to get this beyond just read support?

And I guess 'beyond read support' can be, at a high level be split into 2 parts:
- working inside worktrees, like: stashing, pulling, managing the index, diffing, committing etc 
- worktree management, like: create, delete etc

Context: I'm trying to add worktree support in egit, where I at least want to be able to pull/stash/commit on existing worktrees. 
The management of worktrees would be nice, but can live without that for now

Any thoughts appreciated!

@msohn @jvalkeal

**Comment by msohn** [comment]
I think logical next steps are to add support for creating/deleting worktrees and switching between worktrees in JGit.

**Comment by p-bakker** [comment]
Ok, so is it safe to deduct from your answer that 'working inside worktrees' part  should now already be supported, so when I would expose linked worktrees in EGit, stash/pull/push/commit etc should work on them (from the jgit perspective)?
